src/mpi/init/initthread.c

```diff
@@ -248,6 +248,19 @@ static int thread_cs_init(void)
 
     MPID_THREADPRIV_KEY_CREATE;
 
+    {
+        /*
+         * Trick the Intel compiler into treating MPIR_Per_thread as TLS
+         * See https://github.com/pmodels/mpich/issues/3437
+         */
+        MPIR_Per_thread_t *per_thread = NULL;
+
+        MPID_THREADPRIV_KEY_GET_ADDR(MPIR_ThreadInfo.isThreaded, MPIR_Per_thread_key,
+                                     MPIR_Per_thread, per_thread, &err);
+        MPIR_Assert(err == 0);
+        memset(per_thread->strerrbuf, 0, MPIR_STRERROR_BUF_SIZE);
+    }
```

Comment:
Can we simply do `memset(per_thread, 0, sizeof(MPIR_Per_thread_t);`? Mention of member `strerrbuf` is extra details for this hack. Not mention details also avoids future code change in case we update the details of `MPIR_thread_t`.

Reply:
Yeah, that makes sense. Will fix.